docker-compose.yml

```diff
@@ -0,0 +1,62 @@
+version: '3.8'
+
+services:
+  test_py37:
+    image: python:3.7-slim-bullseye
+    command: sh -c "
+      cd pycaption;
+      pip install --upgrade pip;
+      pip install -r test_requirements.txt;
+      pip install -e .;
+      pytest -vvvv --color=yes --junit-xml=junit.xml --cov=pycaption --cov-report xml:coverage.xml;
+      "
+    volumes:
+      - .:/pycaption
+
+  test_py38:
+    image: python:3.8-slim-bullseye
+    command: sh -c "
+      cd pycaption;
+      pip install --upgrade pip;
+      pip install -r test_requirements.txt;
+      pip install -e .;
+      pytest -vvvv --color=yes --junit-xml=junit.xml --cov=pycaption --cov-report xml:coverage.xml;
+      "
+    volumes:
+      - .:/pycaption
+
+  test_py39:
+    image: python:3.9-slim-bullseye
+    command: sh -c "
+      cd pycaption;
+      pip install --upgrade pip;
+      pip install -r test_requirements.txt;
+      pip install -e .;
+      pytest -vvvv --color=yes --junit-xml=junit.xml --cov=pycaption --cov-report xml:coverage.xml;
+      "
+    volumes:
+      - .:/pycaption
+
+  test_py310:
+    image: python:3.10-slim-bullseye
+    command: sh -c "
+      cd pycaption;
+      pip install --upgrade pip;
+      pip install -r test_requirements.txt;
+      pip install -e .;
+      pytest -vvvv --color=yes --junit-xml=junit.xml --cov=pycaption --cov-report xml:coverage.xml;
+      "
+    volumes:
+      - .:/pycaption
+
+  test_py311:
+    image: python:3.11-slim-bullseye
+    command: sh -c "
+      cd pycaption;
+      pip install --upgrade pip;
+      pip install -r test_requirements.txt;
+      pip install -e .;
+      pytest -vvvv --color=yes --junit-xml=junit.xml --cov=pycaption --cov-report xml:coverage.xml;
```

Comment:
We should record the results as github artifacts if possible.
See https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts